tests/data_classes.py

```diff
@@ -0,0 +1,647 @@
+        if not isinstance(value, tuple):
+            value = (value,)
+
+        # Default to sorting to name always
+        value += (self.name,)
+        return value
+
+    # PARSING FUNCTIONS
+    def full_parse(self):
+        # 10 columns - JID, NAME, BOOT, STATE, TYPE, RELEASE,
+        # IP4, IP6, TEMPLATE, BASEJAIL
+        self.jid, self.name, self.boot, self.state, self.type, \
+        self.release, self.ip4, self.ip6, self.template, self.basejail = \
+            self.standard_parse()
+
+    def all_parse(self):
+        # 4 columns - JID, NAME, STATE, RELEASE, IP4
+        self.jid, self.name, self.state, self.release, self.ip4 = \
+            self.standard_parse()
+
+    def releases_only_parse(self):
+        # 1 column - Bases Fetched
+        self.release = self.standard_parse()[0]
+
+    def quick_parse(self):
+        # 2 column - NAME, IP4
+        # In case of releases, it will only be one # NAME
+        data = self.standard_parse()
+        if len(data) == 1:
+            self.release = data[0]
+        else:
+            self.name, self.ip4 = data
+
+    def df_parse(self):
+        self.name, self.crt, self.res, self.qta, self.use, self.ava = \
+            self.standard_parse()
+
+    def snapshot_parse(self):
+        self.name, self.created, self.rsize, self.used = self.standard_parse()
+
+
+class ZFS:
+    # TODO: Improve how we manage zfs object here
+    pool = None
+    pool_mountpoint = None
+
+    def __init__(self):
+        self.set_pool()
+
+    def set_pool(self):
+        if not self.pool:
+            with libzfs.ZFS() as zfs:
+                pools = [
+                    p for p in zfs.pools
+                    if p.root_dataset.__getstate__().get(
+                        'properties'
+                    ).get(
+                        'org.freebsd.ioc:active', {}
+                    ).get('value', 'no') == 'yes'
+                ]
+
+                if pools:
+                    assert len(pools) == 1, f'{len(pools)} Active pools found'
+
+                    ZFS.pool = pools[0].name
+                    ZFS.pool_mountpoint = pools[0].root_dataset.mountpoint
+
+    @staticmethod
+    def get_dataset(identifier):
+        with libzfs.ZFS() as zfs:
+            return zfs.get_dataset(identifier).__getstate__()
+
+    @staticmethod
+    def get(identifier):
+        with libzfs.ZFS() as zfs:
+            return zfs.get(identifier).__getstate__()
+
+    def _zfs_get_properties(self, identifier):
+        if '/' in identifier:
+            dataset = self.get_dataset(identifier)
+
+            return dataset['properties']
+        else:
+            pool = self.get(identifier)
+
+            return pool['root_dataset']['properties']
+
+    def zfs_get_property(self, identifier, key):
+        try:
+            return self._zfs_get_properties(identifier)[key]['value']
+        except Exception:
+            return '-'
+
+    def get_dataset_safely(self, dataset_name):
```

Comment:
Is there ever a time we don't want to get them safely? Perhaps just remove the other method and rename this one.